src/choleskyQR.jl

```diff
@@ -171,23 +221,75 @@ function resizedata!(K::QRJacobiBand, nm::Integer)
     end
     K
 end
-function cache_filldata!(J::QRJacobiBand{:dv,T}, inds::UnitRange{Int}) where T
+function cache_filldata!(J::QRJacobiBand{:dv,:Q,T}, inds::UnitRange{Int}) where T
+    b = 1+bandwidths(J.U.factors)[2]÷2
+    # pre-fill cached arrays to avoid excessive cost from expansion in loop
+    m, jj = inds[end], inds[2:end]
+    X = jacobimatrix(J.P)[1:m+b+2,1:m+b+2]
+    resizedata!(J.U.factors,m+b,m+b)
+    resizedata!(J.U.τ,m)
+    K, τ, F, dv = J.UX, J.U.τ, J.U.factors, J.data
+    v = Vector{T}(undef,b+3)
+    M = Matrix{T}(undef,b+3,b+3)
+    @inbounds for n in jj
+        v = [zero(T);one(T);F[n+1:n+b,n]]
+        K .= K .- τ[n] .*  v .* (v'K)
+        K .= K .- τ[n] .*  (K*v) .* v'
+        M = Matrix(X[n:n+b+1,n:n+b+1])
+        M[1:end-1,1:end-1] .= K[2:end,2:end]
+        dv[n] = M[1,1] # sign correction due to QR not guaranteeing positive diagonal for R not needed on diagonals since contributions cancel
+        K .= M
+    end
+    J.UX = M
+    J.data[jj] .= dv[jj]
+end
+function cache_filldata!(J::QRJacobiBand{:ev,:Q,T}, inds::UnitRange{Int}) where T
```

Comment:
I had it be in tandem originally in the previous PR but you didn't like my extra struct to accomplish it. I can definitely do that by introducing a helper struct that stores both bands. Shouldn't take long and like I discussed with mikael, this is an obvious 2x speed up.

Reply:
ok I think all your comments are done except this one - this might take a bit longer to make sure it doesn't introduce bugs

Reply:
Hm, maybe we should make sure on what's being considered before I restructure the whole thing, since it'd lead to redundant work.

If you recall, what I had originally was an adaptive $2\times \infty$ cached array that would construct both diagonal and off-diagonal in one swoop. That's nice bc it's fast but it's not that great bc SymTridiagonal takes vectors, not matrices and apparently resizing since vectors is a lot better than resizing matrices. Of course, the current method has the drawback that it recomputes for both bands (roughly a 2x cost increase) but it fits very nicely into SymTridiagonal.

So what is the best struct to go for here? Is there a way to get the best of both worlds or is it back to $2\times \infty$ cached array?

Reply:
Maybe QRJacobiData just stores two vectors that it resizes in tandem?